### ISSUE TITLE
[BMG][Accuracy] timm_models/pit_b_224 AMP inference random fail_accuracy

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#4350]
## Summary

`timm_models/pit_b_224` shows `fail_accuracy` on BMG during AMP inference.

- On nightly, `amp_bf16` passes 10/10 while `amp_fp16` is flaky with 4/10 `fail_accuracy`.
- On release 2.13, both `amp_bf16` and `amp_fp16` become flaky.
- With new Triton (`3.8.0+git9810c442`), both `amp_bf16` and `amp_fp16` fail 10/10.

### nightly 2.14.0.dev20260713+xpu

| dtype | result |
| --- | --- |
| amp_bf16 | 10/10 pass |
| amp_fp16 | 6/10 pass, 4/10 fail_accuracy |

Representative error:

```text
E0715 07:35:34.227000 12897 torch/_dynamo/utils.py:3738] RMSE (res-fp64): 0.02547, (ref-fp64): 0.00042 and shape=torch.Size([8, 1000]). res.dtype: torch.float16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
```

### release 2.13

| dtype | result |
| --- | --- |
| amp_bf16 | 8/10 pass, 2/10 fail_accuracy |
| amp_fp16 | 7/10 pass, 3/10 fail_accuracy |

Representative errors:

```text
E0715 07:12:38.118000 14715 torch/_dynamo/utils.py:3706] RMSE (res-fp64): 0.01911, (ref-fp64): 0.00361 and shape=torch.Size([8, 1000]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
E0715 07:12:38.224000 14714 torch/_dynamo/utils.py:3706] RMSE (res-fp64): 0.07690, (ref-fp64): 0.00043 and shape=torch.Size([8, 1000]). res.dtype: torch.float16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
```

### Triton 3.8.0+git9810c442

| dtype | result |
| --- | --- |
| amp_bf16 | 0/10 pass, 10/10 fail_accuracy |
| amp_fp16 | 0/10 pass, 10/10 fail_accuracy |

Representative errors:

```text
E0715 07:19:26.485000 462072 torch/_dynamo/utils.py:3706] RMSE (res-fp64): 0.02889, (ref-fp64): 0.00361 and shape=torch.Size([8, 1000]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
E0715 07:19:27.034000 462075 torch/_dynamo/utils.py:3706] RMSE (res-fp64): 0.65941, (ref-fp64): 0.00043 and shape=torch.Size([8, 1000]). res.dtype: torch.float16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
```

## Reproducer

Environment setup and repro steps are attached below.

```bash
docker run -it -e TZ=Aisa/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged -v $(realpath ${HOME}):/home/jenkins --shm-size=8g intelgpu/ubuntu-26.04-rolling:26.18 bash

curl -LsSf https://astral.sh/uv/install.sh | sh
source $HOME/.local/bin/env
uv venv myvenv --python 3.12 --clear
source myvenv/bin/activate
uv pip install pip wheel "setuptools<81"

uv pip install torch torchaudio torchvision --index-url https://download.pytorch.org/whl/nightly/xpu --pre

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}

uv pip install pip pandas psutil scipy requests
uv pip install -r .ci/docker/ci_commit_pins/huggingface-requirements.txt
uv pip install timm

rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

python benchmarks/dynamo/timm_models.py -d xpu --accuracy --amp --amp-dtype float16 --inference --only pit_b_224 --backend=inductor --cold-start-latency -n10 --timeout=10800 --disable-cudagraphs
python benchmarks/dynamo/timm_models.py -d xpu --accuracy --amp --amp-dtype bfloat16 --inference --only pit_b_224 --backend=inductor --cold-start-latency -n10 --timeout=10800 --disable-cudagraphs
```

## Expected behavior

`pit_b_224` AMP inference accuracy should be stable on BMG.

## Actual behavior

`pit_b_224` shows intermittent `fail_accuracy` on nightly and release 2.13, then deterministic `fail_accuracy` for both AMP dtypes with Triton `3.8.0+git9810c442`.

## Notes

- This may be related to existing accuracy tracking issues such as #2908 and #4001, but this report isolates a concrete repro on `timm_models/pit_b_224` and shows a worsening trend across builds.
- The strongest signal is the version trend: nightly flake -> release 2.13 broader flake -> new Triton always fails.

**Comment by bjarzemb** [comment]
pit_b_224 was raised earlier here: https://github.com/intel/torch-xpu-ops/issues/2908 . @eryk-roch should we close 4350 as duplicate?

**Comment by eryk-roch** [comment]
@bjarzemb one of them shall be close, please.

**Comment by bjarzemb** [comment]
Let's track pit_b_224 in this issue.

**Comment by bjarzemb** [comment]
@mengfei25 Could you please recheck the fix on Linux? The fix was merged into PyTorch https://github.com/pytorch/pytorch/commit/62a4f51cf34e32503ae138c8f977b434bb90a325.

**Comment by bjarzemb** [comment]
Test passed on Windows 5/5 times with:
PyTorch: fddcbe3 (Jul 28)
torch-xpu-ops: https://github.com/intel/torch-xpu-ops/commit/298168d88308a8039c654f9a64b10782e30c4384 (Jul 24)

**Comment by mengfei25** [comment]
> [@mengfei25](https://github.com/mengfei25) Could you please recheck the fix on Linux? The fix was merged into PyTorch [pytorch/pytorch@62a4f51](https://github.com/pytorch/pytorch/commit/62a4f51cf34e32503ae138c8f977b434bb90a325).

@kaileiyx  Let's verify it with the fixed pytorch

**Comment by kaileiyx** [comment]
Test passed on Linux B60 BMG with latest nightly wheel.
amp_bf16 :  10/10 pass
amp_fp16 :  10/10 pass
PyTorch: e0de08a
